go-hello-nonroot/app.go

```diff
@@ -0,0 +1,87 @@
+package main
+
+import (
+	"crypto/rand"
+	"errors"
+	"fmt"
+	"net"
+	"net/http"
+	"os/user"
+	"time"
+)
+
+var id string
```

Comment:
Seems `id` is only used in `cafeHandler()`.

Can we remove this global, and just call `generateID()` in `cafeHandler()` instead?

Reply:
`id` is defined per server (to identify a "server") . If we call it in `cafeHandler()` the `id` value will change for each call (request received). As we are serving from `main()` and `cafeHandler()` is using `id`... `id` needs to be global.